File: 2020_leetcode/problems/__02__/python/reverse_string.py
def reverseString(str):
    left = 0
    right = len(str) - 1
    while (left < right):
        tmp = str[left]
        str[left] = str[right]
        left +=1
        str[right] = tmp
        right -=1
    print(str)

reverseString(list("hello"))

File: 2020_leetcode/problems/__02__/rust/reverseString.rs

fn reverse_string(mut str: Vec<char>) {
    let mut left = 0usize;
    let mut right = str.len() - 1;
    while left < right {
        let tmp = str[left.clone()];
        str[left.clone()] = str[right];
        left = left + 1;
        str[right.clone()] = tmp;
        right = right - 1;
    }
    println!("{:?}", str);
}

fn main() {
    reverse_string("hello".chars().collect::<Vec<char>>());
}


